src/gui/sdlmain.cpp

```diff
@@ -3571,35 +3580,25 @@ static void HandleMouseWheel(SDL_MouseWheelEvent *wheel)
 
 static void HandleMouseButton(SDL_MouseButtonEvent * button)
 {
-    auto notify_pressed = [](uint8_t button) {
-		switch (button) {
-		case SDL_BUTTON_LEFT:   MOUSE_EventPressed(0); break;
-		case SDL_BUTTON_RIGHT:  MOUSE_EventPressed(1); break;
-		case SDL_BUTTON_MIDDLE: MOUSE_EventPressed(2); break;
-		case SDL_BUTTON_X1:     MOUSE_EventPressed(3); break;
-		case SDL_BUTTON_X2:     MOUSE_EventPressed(4); break;
-		}
-	};
-
-    auto notify_released = [](uint8_t button) {
+    auto notify_button = [](const uint8_t button, const bool pressed) {
 		switch (button) {
-		case SDL_BUTTON_LEFT:   MOUSE_EventReleased(0); break;
-		case SDL_BUTTON_RIGHT:  MOUSE_EventReleased(1); break;
-		case SDL_BUTTON_MIDDLE: MOUSE_EventReleased(2); break;
-		case SDL_BUTTON_X1:     MOUSE_EventReleased(3); break;
-		case SDL_BUTTON_X2:     MOUSE_EventReleased(4); break;
+		case SDL_BUTTON_LEFT:   MOUSE_EventButton(0, pressed); break;
+		case SDL_BUTTON_RIGHT:  MOUSE_EventButton(1, pressed); break;
+		case SDL_BUTTON_MIDDLE: MOUSE_EventButton(2, pressed); break;
+		case SDL_BUTTON_X1:     MOUSE_EventButton(3, pressed); break;
+		case SDL_BUTTON_X2:     MOUSE_EventButton(4, pressed); break;
 		}
 	};
 	
 	if (button->state == SDL_RELEASED) {
-		notify_released(button->button);
+		notify_button(button->button, false);
```

Comment:
Perhaps something like this for improved readability?

`constexpr auto is_pressed = false;`
`notify_button(button->button, is_pressed);`

Reply:
OK, added a `constexpr`.